controllers/profiles/reconciler_prod.go

```diff
@@ -115,28 +122,21 @@ func (h *newBuilderReconciliationState) Do(ctx context.Context, workflow *operat
 	}
 	// If there is an active platform we have got all the information to build but...
 	// ...let's check before if we have got already a build!
-	build, err := buildable.GetWorkflowBuild(workflow.Name, workflow.Namespace)
+	buildManager := builder.NewKogitoServerlessBuildManager(ctx, h.client)
+	build, err := buildManager.GetOrCreateBuild(workflow)
 	if err != nil {
 		return ctrl.Result{}, nil, err
 	}
-	if build == nil {
-		//If there isn't a build let's create and start the first one!
-		build, err = buildable.CreateWorkflowBuild(workflow.Name, workflow.Namespace)
-		if err != nil {
-			return ctrl.Result{}, nil, err
-		}
-	}
-	//If there is a build, let's ask to restart it
-	build.Status.BuildPhase = builderapi.BuildPhaseNone
-	build.Status.Builder.Status = builderapi.BuildStatus{}
-	if err = h.client.Status().Update(ctx, build); err != nil {
-		h.logger.Error(err, fmt.Sprintf("Failed to update Build status for Workflow %s", workflow.Name))
-		return ctrl.Result{}, nil, err
+
+	if build.Status.BuildPhase != operatorapi.BuildPhaseFailed {
+		workflow.Status.Manager().MarkFalse(api.BuiltConditionType, api.BuildIsRunningReason, "")
+		workflow.Status.Manager().MarkFalse(api.RunningConditionType, api.WaitingForBuildReason, "")
+		_, err = h.performStatusUpdate(ctx, workflow)
+	} else {
+		// TODO: not ideal, but we will improve it on https://issues.redhat.com/browse/KOGITO-8792
+		h.logger.Info("Build is in failed state, try to delete the KogitoServerlessBuild to restart a new build cycle")
```

Comment:
will the build always be in the failed state here? isn't better to print the build.Status.BuildPhase instead?

Reply:
Nops, this is a very limited build support. I don't want to add any more features. The phase is printed in the event informer. Don't need to print in the logs too.